serving/src/main/java/feast/serving/service/HistoricalServingService.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2019 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.serving.service;
+
+import feast.serving.ServingAPIProto;
+import feast.serving.ServingAPIProto.*;
+import feast.serving.ServingAPIProto.Job.Builder;
+import feast.serving.specs.CachedSpecService;
+import feast.storage.api.retrieval.FeatureSetRequest;
+import feast.storage.api.retrieval.HistoricalRetrievalResult;
+import feast.storage.api.retrieval.HistoricalRetriever;
+import feast.storage.connectors.bigquery.retrieval.BigQueryHistoricalRetriever;
+import io.grpc.Status;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import org.slf4j.Logger;
+
+public class HistoricalServingService implements ServingService {
+
+  private static final Logger log =
+      org.slf4j.LoggerFactory.getLogger(HistoricalServingService.class);
+
+  private final HistoricalRetriever retriever;
+  private final CachedSpecService specService;
+  private final JobService jobService;
+
+  public HistoricalServingService(
+      HistoricalRetriever retriever, CachedSpecService specService, JobService jobService) {
+    this.retriever = retriever;
+    this.specService = specService;
+    this.jobService = jobService;
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public GetFeastServingInfoResponse getFeastServingInfo(
+      GetFeastServingInfoRequest getFeastServingInfoRequest) {
+    try {
+      BigQueryHistoricalRetriever bigQueryHistoricalRetriever =
+          (BigQueryHistoricalRetriever) retriever;
+      return GetFeastServingInfoResponse.newBuilder()
+          .setType(FeastServingType.FEAST_SERVING_TYPE_BATCH)
+          .setJobStagingLocation(bigQueryHistoricalRetriever.jobStagingLocation())
+          .build();
+    } catch (Exception e) {
```

Comment:
This exception seems too broad. If there is a problem with a retriever then we should be logging it out.